openghg/store/base/_datasource.py

```diff
@@ -123,36 +123,57 @@ def add_timeseries_data(self, data: Dataset) -> None:
             None
         """
         from openghg.util import daterange_overlap
+        from xarray import concat as xr_concat
+        from numpy import unique as np_unique
 
-        # Group by year
-        year_group = data.groupby("time.year")
-        year_data = [data for _, data in year_group if data]
+        # Chunk the data into year datasets by year
+        grouped_by_year = [data for _, data in data.groupby("time.year") if data]
```

Comment:
There's likely to be some conflicts with the `add_bc` PR changes around here after it's merged into devel

Reply:
Ok - yes there's a few conflicts to fix here